### PR TITLE
[py-qscintilla] Fix build on *-osx-dynamic

### PR DESCRIPTION
**Pull request by m-kuhn**

### CODE REVIEW

File: ports/py-qscintilla/portfile.cmake [hunk 1]
```diff
@@ -15,20 +15,26 @@ vcpkg_extract_source_archive(
 
 file(RENAME "${SOURCE_PATH}/Python/pyproject-qt5.toml" "${SOURCE_PATH}/Python/pyproject.toml")
 
+if(VCPKG_TARGET_IS_WINDOWS)
+  set(PY_LIB_DIR "tools/python3/Lib")
```
Comment (m-kuhn) [inline ports/py-qscintilla/portfile.cmake:19]: Questions:

 - why is this different from https://github.com/open-vcpkg/python-registry/pull/32/files#diff-306c19cfe05dc8020e1a5027728c4efe1e8333a5fb2798215db7cd7a908021deR173
 - Should this be exported from python3/vcpkg-port-config.cmake

Comment (Neumann-A) [reply ports/py-qscintilla/portfile.cmake:19]: there is probably a cmake variable in gdal somewhere controlling the install location of python stuff.

File: ports/py-qscintilla/portfile.cmake [hunk 2]
```diff
@@ -15,20 +15,26 @@ vcpkg_extract_source_archive(
...
+else()
+  set(PY_LIB_DIR "lib/python${PYTHON3_VERSION_MAJOR}.${PYTHON3_VERSION_MINOR}")
+endif()
+
 set(SIPBUILD_ARGS
     "--qmake" "${CURRENT_INSTALLED_DIR}/tools/Qt6/bin/qmake${VCPKG_HOST_EXECUTABLE_SUFFIX}"
     "--api-dir" "${CURRENT_PACKAGES_DIR}/share/Qt6/qsci/api/python"
     "--qsci-features-dir" "${SOURCE_PATH}/src/features"
-    "--qsci-include-dir" "${SITE_PACKAGES}/src"
+    "--qsci-include-dir" "${CURRENT_INSTALLED_DIR}/${PY_LIB_DIR}/site-packages/PyQt6/bindings"
```
Comment (Neumann-A) [inline ports/py-qscintilla/portfile.cmake:28]:
```suggestion
    "--qsci-include-dir" "${CURRENT_INSTALLED_DIR}/${PYTHON3_SITE}/PyQt6/bindings"
```
?

File: ports/py-qscintilla/portfile.cmake [hunk 3]
```diff
@@ -15,20 +15,26 @@ vcpkg_extract_source_archive(
...
     "--qsci-library-dir" "${SOURCE_PATH}/src"
     "--no-make"
     "--verbose"
     "--build-dir" "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel"
-    "--target-dir" "${CURRENT_INSTALLED_DIR}/tools/python3/Lib/site-packages/"
+    "--target-dir" "${CURRENT_INSTALLED_DIR}/${PY_LIB_DIR}/site-packages/"
```
Comment (Neumann-A) [inline ports/py-qscintilla/portfile.cmake:33]:
```suggestion
    "--target-dir" "${CURRENT_INSTALLED_DIR}/${PYTHON3_SITE}"
```
?

File: ports/py-qscintilla/vcpkg.json [hunk 4]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "py-qscintilla",
```
Comment (Neumann-A) [inline ports/py-qscintilla/vcpkg.json:2]: intended to not depend on python3 ?

**Comment by Neumann-A** [comment]
If https://github.com/microsoft/vcpkg/pull/39449 is merged make sure to depend on the [extensions] feature. Also do we need to change  "vcpkg-python-scripts" to auto pull in the `vcpkg-port-config` of `python3`?

File: ports/py-qscintilla/vcpkg.json (hunk 4 above)
Comment (m-kuhn) [reply ports/py-qscintilla/vcpkg.json:2]: indirectly via py-pyqt6, is this not ideal?

Comment (Neumann-A) [reply ports/py-qscintilla/vcpkg.json:2]: Depends if another script port is able to load the port config from python3. But the python dependency should probably always be explicit.

**Comment by m-kuhn** [comment]
Thank you for the review @Neumann-A

**Pull request merged**